guides/micronaut-k8s-gcp/micronaut-k8s-gcp.adoc

```diff
@@ -0,0 +1,188 @@
+
+You will discover how the Micronaut framework eases Kubernetes integration and deployment to GKE.
+
+common:costs-gcloud.adoc[]
+
+external:micronaut-k8s/requirements.adoc[]
+common:jq[]
+* You need a https://cloud.google.com/gcp/[Google Cloud Platform (GCP)] account and a GCP project.
+
+external:micronaut-k8s/create-app-external-guides.adoc[]
+
+common:gcp.adoc[]
+
+common:gcp-billing.adoc[]
+
+common:gcp-docker.adoc[]
+
+=== Create GKE cluster
+
+In this guide we will create cluster with name `micronaut-k8s` and we will use 2x `n1-standard-2` instances for GKE nodes. Everything will be deployed on your default Google Cloud region. You can check your default region by running:
+
+[source, bash]
+----
+> gcloud config get-value compute/region
+----
+
+Create `micronaut-k8s` kubernetes cluster:
+
+[source, bash]
+----
+gcloud container clusters create micronaut-k8s --machine-type n1-standard-2 --num-nodes 2
+----
+
+== Prepare and Deploy Microservices
+
+Define `GCP_PROJECT_ID` environment variable to store your GCP project id.
+
+[source,bash]
+----
+export GCP_PROJECT_ID="$(gcloud config get-value project)"
+----
+
+=== Users Microservice
+
+external-template:micronaut-k8s-gcp/export-repository.adoc[arg0=users]
+external-template:micronaut-k8s-gcp/k8s-microservice.adoc[arg0=users]
+external-template:micronaut-k8s-gcp/docker-microservice.adoc[arg0=users]
+
+=== Orders Microservice
+
+external-template:micronaut-k8s-gcp/export-repository.adoc[arg0=orders]
+external-template:micronaut-k8s-gcp/k8s-microservice.adoc[arg0=orders]
+external-template:micronaut-k8s-gcp/docker-microservice.adoc[arg0=orders]
+
+=== API Microservice
+
+external-template:micronaut-k8s-gcp/export-repository.adoc[arg0=api]
+external-template:micronaut-k8s-gcp/k8s-microservice.adoc[arg0=api]
+external-template:micronaut-k8s-gcp/docker-microservice.adoc[arg0=api]
+
+=== Deploy Services to GKE
+
+Create a directory for `kubectl` configuration.
+
+[source,bash]
+----
+mkdir -p $HOME/.kube
+----
+
+Install kubectl component.
+
+[source, bash]
+----
+gcloud components install kubectl
+----
+
+Generate `kubectl` configuration for authentication to GKE.
+
+[source, bash]
+----
+gcloud container clusters get-credentials micronaut-k8s
+----
+
+Set `KUBECONFIG` to the created config file. This variable is consumed by `kubectl`.
+
+[source,bash]
+----
+export KUBECONFIG=$HOME/.kube/config
+----
+
+Deploy the _auth.yml_ file that we created in the guideLink:micronaut-k8s[Kubernetes and the Micronaut Framework] guide.
+
+[source,bash]
+----
+kubectl apply -f auth.yml
+----
+
+Create `gcpsersecret` secret that will be used for authentication to Google Cloud Container Registry. This is needed because GKE needs credentials to be able to pull microservices images.
```

Comment:
```suggestion
Create a secret named `gcpsersecret` that will be used for authentication to Google Cloud Container Registry. This is needed because GKE needs credentials to be able to pull microservices images.
```